### PR TITLE
Front end fix

### PR DESCRIPTION
**Pull request by Plypy**
Just having the form validation moved to frontend.
To be completed.

### CODE REVIEW

**Comment by dkayiwa** [comment]
Can you squash the commits into one?

**Comment by Plypy** [comment]
Actually, that's just a reminder for robby to check this...

**Comment by robbyoconnor** [comment]
Don't merge this @dkayiwa

**Comment by robbyoconnor** [comment]
It's really not a good idea to ban code here unless you know it works -- in this case some of our tests are borked :stuck_out_tongue:  @Plypy

**Comment by robbyoconnor** [comment]
It's really not a good idea to just merge code here unless you know it works -- in this case some of our tests are borked :stuck_out_tongue:  @Plypy

**Comment by robbyoconnor** [comment]
@dkayiwa -- also while I DO typically like one tidy commit -- this is fine.

**Comment by robbyoconnor** [comment]
There's one failing test it seems :)

**Comment by Plypy** [comment]
Yes, it failed, because I haven't run the tests on my machine. But the test that failed simply don't matter at all, just some middleware the new validation process no longer needed. Anyway, I'll fix that.

But really, Robby, how do you think of the view? We're moving very slow recently.

**Comment by robbyoconnor** [comment]
I'll run it now -- then do the work I was meaning to do -- was focusing on prepping for my interview but now have more time...will move faster!

**Comment by robbyoconnor** [comment]
Okay sorry for the delay -- I looked 

# Sign up page issues:

image

I'm not crazy with the fact that the error text appears over the textbox -- it looks messy -- also the color of the button is meh...let's choose one of the openmrs colors instead -- I hate that color -- **REALLY** hate it. The error message however is somewhat useful but from a UX POV it doesn't make sense...

File: app/validate.js [hunk 1]
```diff
@@ -100,7 +93,7 @@ validate.chkEmailInvalidOrDup = function (email, callback) {
       }
       if (user) {
         // duplicate
-        return callback(null, EMAIL_DUP_MSG);
+        return callback(null, 'Already used');
       }
```
Comment (robbyoconnor) [inline app/validate.js]: No, go back to the constant.

File: app/validate.js [hunk 2]
```diff
@@ -68,7 +61,7 @@ validate.chkUsernameInvalidOrDup = function (username, callback) {
     }
     if (user) {
       // duplicate
-      return callback(null, USERNAME_DUP_MSG);
+      return callback(null, 'Already used');
     }
```
Comment (robbyoconnor) [inline app/validate.js:66]: No, go back to the constant.

File: app/validate.js [hunk 3]
```diff
@@ -109,7 +102,7 @@ validate.chkEmailInvalidOrDup = function (email, callback) {
 
 validate.chkPassword = function (password, passhash, callback) {
   if (!utils.checkSSHA(password, passhash)) {
-    return callback(null, WRONG_PWD_MSG);
+    return callback(null, 'Wrong');
   }
```
Comment (robbyoconnor) [inline app/validate.js:107]: `Wrong password` would be better ;)

File: resource/less/colors.less [hunk 4]
```diff
@@ -1,7 +1,10 @@
...
 // Copied from Bootstrap config in /resource/bootstrap/config.json:
 @brand-primary: #5B57A6;
 @brand-success: #009384;
-@brand-info: #5bc0de;
+@brand-info:    #5bc0de;
 @brand-warning: #EEA616;
-@brand-danger: #F26522;
\ No newline at end of file
+@brand-danger:  #F26522;
+
```
Comment (robbyoconnor) [inline resource/less/colors.less:15]: Go back to the color-scheme we were using before, I liked it.

**Comment by robbyoconnor** [comment]
Also, unrelated but we something that could be handled with this a [js injection bug exists](https://issues.openmrs.org/browse/ID-126).

File: app/validate.js (hunk 2 above)
Comment (robbyoconnor) [inline app/validate.js:66]: The rationale is I want things DRY

**Comment by robbyoconnor** [comment]
Password Reset page seems broken...the validation manages seem to show...was it meant to be a dummy form? 

image

**Comment by robbyoconnor** [comment]
Looked at the rest,Color scheme should go back to what it was --  I liked the purple :)  Definitely **will not** merge as-is, sorry :frowning:

**Comment by Plypy** [comment]
> Definitely will not merge as-is, sorry :frowning:

That's OK, I never intended to have it merged right now.

**Comment by Plypy** [comment]
Had been in Spring Festival state. :smile: 

I've fixed some bugs and completed those forms. And the new color of buttons that I've used is chosen from the upper part of OpenMRS logo, named as `omrs-red` in colors.less. (It may not be red exactly, I'm not sensitive to distinguishing colors, namely I have some color weakness)

Hope you like it now, and I'll fix the failing tests later.

**Comment by robbyoconnor** [comment]
OpenMRS doesn't use red, they use orange, [here](https://wiki.openmrs.org/display/RES/OpenMRS+Logo+Policy) are all the colors.

File: resource/less/colors.less [hunk 5]
```diff
@@ -1,7 +1,15 @@
 
+// openmrs colors
+@omrs-red: #f26622;
```
Comment (robbyoconnor) [inline resource/less/colors.less]: This is orange

File: resource/less/colors.less [hunk 6]
```diff
@@ -1,7 +1,15 @@
...
+@omrs-orange: #eea716;
+
```
Comment (robbyoconnor) [inline resource/less/colors.less:5]: This is gold

Comment (robbyoconnor) [inline resource/less/colors.less]:
```less
@omrs-gold: #eea716;
```

Comment (robbyoconnor) [inline resource/less/colors.less:5]:
``` less
@omrs-gold: #eea716;
```

File: resource/less/colors.less (hunk 5 above)
Comment (Plypy) [inline resource/less/colors.less]: ...well orange it is...

And the 'omrs-orange' shall be gold  as well.

Damned, I know these could be wrong...

Comment (robbyoconnor) [inline resource/less/colors.less]:
``` less
@omrs-orange: #f26622;
```

**Comment by Plypy** [comment]
fixed :smiley:

**Comment by robbyoconnor** [comment]
@Plypy I will look at it soon. :heart:

**Comment by Plypy** [comment]
> @Plypy I will look at it soon. :heart:

Days passed. :disappointed:

**Comment by ghost** [comment]
:sleeping: :zzz: 

;-)

**Comment by robbyoconnor** [comment]
I'm not paid for this! It's basically whenever I have time! I do feel bad for making you wait...it will be this week -- I promise this time!

**Comment by robbyoconnor** [comment]
Going to review this now

**Comment by robbyoconnor** [comment]
Merging this -- I have to do it by hand as I made changes locally

**Comment by robbyoconnor** [comment]
I merged this and squashed them into one commit.

**Comment by robbyoconnor** [comment]
Squashed and merged in d9988ca , made the script task the default task in 99ed3f9ef6d579fe1bcdc2d6e5da4b760e93ed5c

**Pull request closed without merging**